zulipterminal/model.py

```diff
@@ -516,6 +516,45 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def prevent_compose_box_in_stream(self, stream_identifier: Any) -> bool:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and allow/restrict the compose box from opening and accordingly
+        update the footer
+        """
+        if self.stream_dict[stream_identifier].get(
+            "stream_post_policy"
+        ) == 2 or self.stream_dict[stream_identifier].get("is_announcement_only"):
```

Comment:
You've done this, but note that you've replaced a `.get` with an index; this may or may not be a problem, but this will be clearer once you have test cases for different server versions.